### DISCUSSION TITLE
How to view orphaned resources?

### DISCUSSION THREAD

**Discussion by roy-work** [argoproj/argo-cd#13063] [Q&A] [answered]
[The docs](https://argo-cd.readthedocs.io/en/stable/user-guide/orphaned-resources/) say,

> While warning disabled, application users can still view orphaned resources in the UI.

Even _with_ the warning enabled, I'm not able to figure out how to do this? The warning appears in the project, and tells me there are orphaned resources, but it doesn't say what those are.

It also says you can "inspect/remove" them via the UI, which I'd also like to do.

I should also mention the docs state,
> The orphaned resources can be located using the application details page:

And then appear to show an application details page showing an orphaned resource.

This makes little semantic sense (orphans are part of a _project_, not an application, no? The docs themselves even say as much: "Orphaned Kubernetes resource is a top-level namespaced resource which does not belong to any Argo CD Application" — what app's details page am I suppose to even be looking at?), but it also doesn't work: my application details page has the warning, but does not show any orphaned resources.

(Relatedly, I'm also not clear on how Argo detects orphans. I would have expected resources managed by Argo to have an annotation to that effect, but they do not. There is `app.kubernetes.io/instance` but that seemed unlikely, given the potential for collision.)

**Comment by roy-work** [comment, marked as answer]
One needs to click "Show Orphaned" on the sidebar.

I still wonder a bit about the UI here, as these are not logically associated with any application. This seems like Projects should be given a more prominent place in the UI, and orphans are a subset of those.

It appears the other answer to my question is that it's _any_ resource in the namespace, with some exceptions noted by the docs. That also explains the rather odd "Orphaned Kubernetes resource is a top-level **namespaced** resource" — after all, why would an orphan need to be namespaced? One could easily orphan non-namespaced resources.

These are more like "unmanaged" — they might have been orphaned, or they might just be outside Argo. (Argo seems to presume here that it owns the NS, which isn't all that unreasonable.)

**Reply by jsoref** [reply]
Fwiw, I have a few tickets to try to improve the experience
* https://github.com/argoproj/argo-cd/issues/5147
* https://github.com/argoproj/argo-cd/issues/5158
* https://github.com/argoproj/argo-cd/issues/21356
* https://github.com/argoproj/argo-cd/issues/21357

**Comment by sourcehawk** [comment]
ArgoCD shows this message 
<img width="1322" height="760" alt="image" src="https://github.com/user-attachments/assets/38e4b0ce-1234-4716-9d27-97bafe67fa8d" />

Why doesn’t the message include the name of the orphaned resource?

I can’t figure out how to identify it in the resource graph. Toggling Show orphaned on and off and trying to spot the difference in a graph that doesn’t fit on one page is honestly a pretty terrible UI experience 😅

Is there another way to view a list of the orphaned resources that I’m missing?
